Kernel/Devices/USB/USBTransfer.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Devices/USB/USBTransfer.h>
+#include <Kernel/VM/MemoryManager.h>
+
+namespace Kernel::USB {
+
+RefPtr<USBTransfer> USBTransfer::create(const USBPipe& pipe, u16 len)
+{
+    auto vmobj = ContiguousVMObject::create_with_size(PAGE_SIZE);
+    if (!vmobj)
+        return nullptr;
+
+    return adopt_ref_if_nonnull(new USBTransfer(pipe, len, vmobj));
+}
+
+USBTransfer::USBTransfer(const USBPipe& pipe, u16 len, RefPtr<ContiguousVMObject>& vmobj)
```

Comment:
`RefPtr<ContiguousVMObject>&` implies that this function wants to mutate the `RefPtr` provided by the caller.
What we really want here is simply `ContiguousVMObject&` since that's enough for passing along to `allocate_kernel_region_with_vmobject` anyway :)